elasticsearch/src/test/java/org/apache/calcite/adapter/elasticsearch/ElasticSearchAdapterTest.java

```diff
@@ -705,6 +737,97 @@ public void approximateCount() throws Exception {
+   *
+   * @throws Exception
+   */
+  @Test
+  public void testMatchQuery() throws Exception {
+
+    final Map<String, String> mapping = ImmutableMap.of("city", "text", "state",
+        "keyword", "pop", "long");
+    setupInstanceWithMapping(mapping, ZIPS_WITH_TEXT);
+
+    try {
+      CalciteConnection con = (CalciteConnection) newConnectionFactoryForCustomIndex
+          (ZIPS_WITH_TEXT).createConnection();
+      SchemaPlus postschema = con.getRootSchema().getSubSchema("elastic");
+
+      FrameworkConfig postConfig = Frameworks.newConfigBuilder()
+          .parserConfig(SqlParser.Config.DEFAULT)
+          .defaultSchema(postschema)
+          .build();
+
+      final RelBuilder builder = RelBuilder.create(postConfig);
+      builder.scan(ZIPS_WITH_TEXT);
+
+      final RelDataTypeFactory relDataTypeFactory = new SqlTypeFactoryImpl(
+          RelDataTypeSystem.DEFAULT);
+      final RexBuilder rexbuilder = new RexBuilder(relDataTypeFactory);
+
+      RexNode nameRexNode = rexbuilder.makeCall(SqlStdOperatorTable.ITEM,
+          rexbuilder.makeInputRef(relDataTypeFactory.createSqlType(SqlTypeName.ANY), 0),
+          rexbuilder.makeCharLiteral(
+            new NlsString("city", rexbuilder.getTypeFactory().getDefaultCharset().name(),
+            SqlCollation.COERCIBLE)));
+
+      RelDataType mapType = relDataTypeFactory.createMapType(
+          relDataTypeFactory.createSqlType(SqlTypeName.VARCHAR),
+          relDataTypeFactory.createTypeWithNullability(
+            relDataTypeFactory.createSqlType(SqlTypeName.ANY), true));
+
+      ArrayList<RexNode> namedList = new ArrayList<RexNode>(2);
+      namedList.add(rexbuilder.makeInputRef(mapType, 0));
+      namedList.add(nameRexNode);
+
+      //Add fields in builder stack so it is accessible while filter preparation
+      builder.projectNamed(namedList, Arrays.asList("_MAP", "city"), true);
+
+      RexNode filterRexNode = builder
+          .call(SqlStdOperatorTable.CONTAINS, builder.field("city"),
+          builder.literal("waltham"));
+      builder.filter(filterRexNode);
+
+      String builderExpected = ""
+          + "LogicalFilter(condition=[CONTAINS($1, 'waltham')])\n"
+          + "  LogicalProject(_MAP=[$0], city=[ITEM($0, 'city')])\n"
+          + "    LogicalTableScan(table=[[elastic, zipswithtext]])\n";
+
+      RelNode root = builder.build();
+
+      RelRunner ru = (RelRunner) con.unwrap(Class.forName("org.apache.calcite.tools.RelRunner"));
+      try (PreparedStatement preparedStatement = ru.prepare(root)) {
+
+        String s = CalciteAssert.toString(preparedStatement.executeQuery());
+        final String result = ""
+            + "_MAP={id=02154, city=NORTH WALTHAM, loc=[-71.236497, 42.382492], pop=57871, state=MA}; city=NORTH WALTHAM\n";
+
+        //Validate query prepared
+        assertThat(root, hasTree(builderExpected));
+
+        //Validate result returned from ES
+        assertThat(s, is(result));
+      }
+    } catch (Exception e) {
```

Comment:
You don't need to wrap the exception. Let it propagate and be reported by JUnit

Reply:
Done